config/initializers/opentelemetry.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Set OTEL_* environment variables according to OTel docs:
+# https://opentelemetry.io/docs/concepts/sdk-configuration/
+#
+# Totally hypothetically, if one wanted to send traces to Honeycomb,
+# OTEL_EXPORTER_OTLP_ENDPOINT=https://api.honeycomb.io
+# OTEL_EXPORTER_OTLP_HEADERS="x-honeycomb-team=<apikey>"
+
+# If an OTEL endpoint of some variety has been set,
+# light things up.
+if ENV.keys.any? {|name| name.match?(/OTEL_.*_ENDPOINT/)}
+  # required for the crafting of telemetry
+  require 'opentelemetry/sdk'
+  # required for the sending of telemetry
+  require 'opentelemetry/exporter/otlp'
+  # convenience to include all auto-instrumentations
+  # available from OTel Ruby contrib
+  require 'opentelemetry/instrumentation/all'
+
+
+  OpenTelemetry::SDK.configure do |c|
+    # set a sensible service name if one wasn't provided
+    # by environment variable
+    unless ENV['OTEL_SERVICE_NAME'].present?
+      c.service_name = case $PROGRAM_NAME
+                        when /puma/
+                          "web"
```

Comment:
I feel it would require always have `OTEL_SERVICE_NAME`.
I would propose to name it as `mastodon_server` and `mastodon_jobs` accordingly.

Reply:
> I feel it would require always have `OTEL_SERVICE_NAME`.

Are you saying that ... 

1. ... you predict that this code will result in `OTEL_SERVICE_NAME` always being required?

According to my testing, it does not. `OTEL_SERVICE_NAME` is optional.

Which was handy for @jgaskins, who could set it when running things like `rails db:migrate`.

2. ... you recommend that `OTEL_SERVICE_NAME` become required to enable OTel instrumentation and the environment variable must be set outside the Rails processes?

I recommend it remain optional. Providing default service names avoids adding more complexity to running a Mastodon instance. In determining the service name programmatically, I am very open to finding a way that is better than interrogating `$PROGRAM_NAME`.

> I would propose to name it as `mastodon_server` and `mastodon_jobs` accordingly.

Different names _could_ be useful, but I suggest not at the moment.

I opted for names that map one-to-one to the names given to these processes in the [Procfile](https://github.com/mastodon/mastodon/blob/v3.5.3/Procfile.dev#L1-L2) and in docker-compose ([web](https://github.com/mastodon/mastodon/blob/v3.5.3/docker-compose.yml#L45), [sidekiq](https://github.com/mastodon/mastodon/blob/v3.5.3/docker-compose.yml#L84)) to reduce confusion for an operator trying to correlate what they see in telemetry to the names of the things they are running.

Reply:
@miry Thank you for the feedback! It made me think and write things down. Probably some of those words should go in the PR description and/or code comments. 💞

Reply:
More words added in 7a729bb62ee7f02c15fd79d419c4011c9ea57673. Thanks!